src/components/swap/routing/hooks/useTrade.ts

```diff
@@ -296,3 +306,161 @@ export function useUbeswapTradeExactOut(tokenIn?: Token, tokenAmountOut?: TokenA
     return bestTrade
   }, [tokenIn, tokenAmountOut, allowedPairs, singleHopOnly, directTrade, disableSmartRouting, moolaRoute])
 }
+
+interface Dependencies {
+  chainId: ChainId
+  account: string | null
+  allowedSlippage: number
+  singleHopOnly: boolean
+  inputAddr: string | undefined
+  outputAddr: string | undefined
+  inputAmount: string | undefined
+}
+
+export function useMinimaTrade(tokenAmountIn?: TokenAmount, tokenOut?: Token): MinimaRouterTrade | null {
+  const [minimaTrade, setMinimaTrade] = React.useState<MinimaRouterTrade | null>(null)
+  const [deps, setDeps] = React.useState<Dependencies | undefined>(undefined)
+  const [singleHopOnly] = useUserSingleHopOnly()
+  const [allowedSlippage] = useUserSlippageTolerance()
+  const [fetchUpdatedData, setFetchUpdatedData] = React.useState<boolean>(true)
+  const [fetchTimeout, setFetchTimeout] = React.useState<NodeJS.Timeout | undefined>(undefined)
+  const { address: account, network } = useContractKit()
+  const { chainId } = network
+  const library = useProvider()
+  const provider = getProviderOrSigner(library, account || undefined)
+  const tokens = useAllTokens()
+  const call = React.useCallback(async () => {
+    if (!tokenAmountIn?.currency.address || !tokenAmountIn?.raw || !tokenOut?.address) {
+      setMinimaTrade(null)
+      setDeps(undefined)
+      return
+    }
+    const curDeps = {
+      chainId,
+      account,
+      allowedSlippage,
+      singleHopOnly,
+      inputAddr: tokenAmountIn.currency.address,
+      outputAddr: tokenOut.address,
+      inputAmount: tokenAmountIn.raw.toString(),
+    }
+    if (_.isEqual(deps, curDeps) && !fetchUpdatedData) {
+      return
+    }
+    if (!fetchUpdatedData) {
+      setMinimaTrade(null)
+    }
+    setDeps(curDeps)
+    setFetchUpdatedData(false)
+    // fetch information of minima router
+    await fetch(
+      `${MINIMA_API_URL}?tokenIn=${tokenAmountIn?.currency.address ?? ''}&tokenOut=${
+        tokenOut?.address ?? ''
+      }&amountIn=${tokenAmountIn?.raw}&slippage=${allowedSlippage}&maxHops=${
+        singleHopOnly ? 1 : MAX_HOPS
+      }&includeTxn=true&priceImpact=1`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-API-KEY': MINIMA_API_KEY,
+        },
+      }
+    )
+      .then(async (res) => {
+        if (res.status !== 200) {
+          setMinimaTrade(null)
+          return
+        }
+        await res
+          .json()
+          .then(async (data: MinimaTradePayload) => {
+            if (data.details) {
+              const path = await Promise.all(
+                data.details.path.map(async (path) => {
```

Comment:
you are shadowing the `path` variable. Makes it harder to follow this code. Considering using a different variable name here like `pathItem`